lib/storage-service.js

```diff
@@ -66,18 +66,28 @@ class StorageService {
    */
   lookup(token, secret, ProviderType) {
     return this._getLeaseManager(token, secret, ProviderType).then((manager) => {
+      const data = {
+        correlation_id: manager.correlation_id
+      };
+
       if (manager.status === STATUS.READY) {
+        StorageService.logManagerEvent('INFO', 'Manager was able to lookup a secret.', manager);
+
         // LeaseManager is ready, immediately resolve with data
-        return manager.data;
+        return Object.assign(data, manager.data);
       }
 
       // Queue the promise states with a timeout
-      return onceWithTimeout(manager, 'ready', this.timeout)
-        .then(() => manager.data)
-        .catch((err) => {
-          this._managers.delete(manager.name);
-          throw err;
-        });
+      return onceWithTimeout(manager, 'ready', this.timeout).then(() => {
+        StorageService.logManagerEvent('INFO', 'Manager was able to lookup a secret.', manager);
+
+        return Object.assign(data, manager.data);
+      })
+      .catch((err) => {
+        StorageService.logManagerEvent('ERROR', err, manager);
+        this._managers.delete(manager.name);
```

Comment:
Is it okay if `this. _getLeaseManager` throws and the error's caught here? Will the `manager` variable be defined, or will the `manager.name` call fail?